src/lib.rs

```diff
@@ -383,6 +383,51 @@
 //! # }
 //! ```
 //!
+//! ## Inspecting errors
+//!
+//! An error-chain error contains information about the error itself, a backtrace, and the chain
+//! of causing errors. For reporting purposes, this information can be accessed as follows.
+//!
+//! ```
+//! # #[macro_use] extern crate error_chain;
+//! use error_chain::ChainedError;  // for e.display()
+//!
+//! error_chain! {
+//!     errors {
+//!         InvalidToolchainName(t: String) {
+//!             description("invalid toolchain name")
+//!             display("invalid toolchain name: '{}'", t)
+//!         }
+//!     }
+//! }
+//!
+//! # fn main() {
+//! let cause = "xyzzy".parse::<i32>().err().unwrap();
+//! let e = Error::with_chain(cause, ErrorKind::InvalidToolchainName("xyzzy".to_string()));
```

Comment:
Maybe you could use `"x".parse::<i32>().chain_err(...).unwrap_err()`? It would closer to real world code.